python/pyspark/sql/functions.py

```diff
@@ -1637,6 +1637,27 @@ def explode(col):
     return Column(jc)
 
 
+@since(2.1)
+def posexplode(col):
```

Comment:
cc @rxin , is `posexplode` a special hive fallback function that we need to register? other ones don't get registered in `functions`